src/go/serviceconfig/service_config_fetcher_test.go

```diff
@@ -0,0 +1,171 @@
+
+	confpb "google.golang.org/genproto/googleapis/api/serviceconfig"
+	smpb "google.golang.org/genproto/googleapis/api/servicemanagement/v1"
+)
+
+func initServiceManagementForTestServiceConfigFetcher(t *testing.T,
+	serviceRollouts *smpb.ListServiceRolloutsResponse, serviceConfig *confpb.Service, serviceName string) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case fmt.Sprintf("/v1/services/%s/rollouts", serviceName):
+			resp, err := proto.Marshal(serviceRollouts)
+			if err != nil {
+				t.Fatalf("fail to generate rollout response: %v", err)
+			}
+			_, _ = w.Write(resp)
+		case fmt.Sprintf("/v1/services/%s/configs/%s", serviceName, serviceConfig.Id):
+			resp, err := proto.Marshal(serviceConfig)
+			if err != nil {
+				t.Fatalf("fail to generate config response: %v", err)
+			}
+			_, _ = w.Write(resp)
+		default:
+			fmt.Printf(r.URL.Path)
+			w.WriteHeader(http.StatusNotFound)
+		}
+
+	}))
+}
+
+func genRolloutAndConfig(serviceRolloutId, serviceConfigId string) (*smpb.ListServiceRolloutsResponse, *confpb.Service) {
+	serviceRollouts := &smpb.ListServiceRolloutsResponse{
+		Rollouts: []*smpb.Rollout{
+			{
+				RolloutId: serviceRolloutId,
+				Strategy: &smpb.Rollout_TrafficPercentStrategy_{
+					TrafficPercentStrategy: &smpb.Rollout_TrafficPercentStrategy{
+						Percentages: map[string]float64{
+							serviceConfigId: 100,
+						},
+					},
+				},
+			},
+		},
+	}
+	serviceConfig := &confpb.Service{
+		Name: "foo",
+		Id:   serviceConfigId,
+	}
+	return serviceRollouts, serviceConfig
+}
+
+func TestServiceConfigFetcherFetchConfig(t *testing.T) {
+	serviceName := "service-name"
+	serviceRolloutId := "test-rollout-id"
+	serviceConfigId := "test-config-id"
+	serviceRollout, serviceConfig := genRolloutAndConfig(serviceRolloutId, serviceConfigId)
+
+	serviceManagementServer := initServiceManagementForTestServiceConfigFetcher(t, serviceRollout, serviceConfig, serviceName)
+	accessToken := func() (string, time.Duration, error) { return "access-token", time.Duration(60), nil }
+
+	scf := NewServiceConfigFetcher(&http.Client{}, serviceManagementServer.URL, "service-name", accessToken)
+
+	_test := func(configId string, wantServiceConfig *confpb.Service, wantError string) {
+		getConfig, err := scf.FetchConfig(configId)
+		if err != nil {
+			if wantError == "" {
+				t.Fatalf("fail to fetch config: %v", err)
+			}
+
+			if err.Error() != wantError {
+				t.Fatalf("want error: %s, get error: %v", wantError, err)
+			}
+			return
+		}
+
+		if getConfig == nil {
+			if wantServiceConfig != nil {
+				t.Fatalf("want service config: %v, get service config: nil", wantServiceConfig)
+			}
+			return
+		}
+
+		if !proto.Equal(getConfig, serviceConfig) {
+			t.Fatalf("want service config: %v, get service config: %v", serviceConfig, getConfig)
+		}
+	}
+
+	// Test success of fetching the service config.
+	_test(serviceConfigId, serviceConfig, "")
```

Comment:
These (and other tests) don't seem to follow table-driven golang test style? Could you refactor? Code should be pretty much the same.